drivers/cache/cache_kinetis.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/cache.h>
+
+#include "fsl_cache.h"
+
+void cache_data_enable(void)
+{
+	L1CACHE_EnableSystemCache();
+}
+
+void cache_data_disable(void)
+{
+	L1CACHE_DisableSystemCache();
+}
+
+void cache_instr_enable(void)
+{
+	L1CACHE_EnableCodeCache();
+}
+
+void cache_instr_disable(void)
+{
+	L1CACHE_DisableCodeCache();
+}
+
+int cache_data_all(int op)
+{
+	if (op == K_CACHE_INVD) {
+		L1CACHE_CleanInvalidateSystemCache();
+	} else {
+		return -ENOTSUP;
+	}
+	return 0;
+}
```

Comment:
I'm not blocking this PR but just an heads up that I have already something ready to address #50747 so we have two choices here:
- if ACKed we push this and I'll rework it to be compliant to the new API
- this PR is blocked on #50747

BTW, having your comments on #50747 about what you want to see in the new API would be very valuable given that not many people are writing drivers for external controllers.

Reply:
@carlocaione I made the changes you requested, does it seem okay to you?

Reply:
Please, fix the CI issues before I take a look.